sharedbuffers/mapped_struct.py

```diff
@@ -1001,6 +1001,166 @@ def __str__(self):
     setattr(proxied_list, '__eq__', getattr(proxied_list, '_eq'))
     setattr(proxied_list, '__ne__', getattr(proxied_list, '_ne'))
 
+@cython.cclass
+class proxied_frozenset(object):
+
+    cython.declare(objlist = object)
+
+    def __init__(self, objlist):
+        self.objlist = objlist
+
+    @classmethod
+    def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
+        return mapped_frozenset.pack_into(obj, buf, offs, idmap, implicit_offs, sort=True)
+
+    @classmethod
+    @cython.locals(
+        i=int, j=int, offs=cython.longlong,
+        pybuf='Py_buffer', pbuf='const unsigned char *', b=cython.uchar)
+    def unpack_from(cls, buf, offs, idmap = None):
+        buf = _likerobuffer(buf)
+        if cython.compiled:
+            PyObject_GetBuffer(buf, cython.address(pybuf), PyBUF_SIMPLE)
+            pbuf = cython.cast(cython.p_uchar, pybuf.buf)
+            if offs >= pybuf.len:
+                PyBuffer_Release(cython.address(pybuf))
+                raise IndexError("Offset out of range")
+        else:
+            pbuf = buf
+        try:
+            if pbuf[offs] == 'm':
+                # inline bitmap
+                if cython.compiled and offs+7 >= pybuf.len:
+                    raise IndexError("Object spans beyond buffer end")
+                rv = []
+                for i in xrange(7):
+                    b = ord(pbuf[offs+1+i])
+                    if b:
+                        for j in xrange(8):
+                            if b & (1<<j):
+                                rv.append(i*8+j)
+                return frozenset(rv)
```

Comment:
You should create an object that takes the buffer and queries the bitmap for `__contains__` rather than fleshing out the frozenset. In essence, this isn't a proxied set. This is a regular mapped_frozenset here.

Reply:
That's true. I did it that way because I felt that for such short frozensets, allocating a fresh one didn't make much of a difference.

Reply:
I guess it depends, but since you've got access to the buffer already, I would expect instantiating a proxy would be less overhead than instantiating a full set. Also, lookups in a bitmap are O(1).